addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1540,8 +1567,6 @@ const ColorpickerUserValueWidget = SelectUserValueWidget.extend({
             this._ccValue = color;
         }
 
-        await this._colorPaletteRenderPromise;
```

Comment:
What prevents running the code after this before the colorpalette is fully rendered? :thinking:

Reply:
Now `this._renderColorPalette();` is awaited in `open();` so it should be ready if the dropdown is open/clicked before this point.

Reply:
> so it should be ready if the dropdown is open/clicked before this point.

The question is why is that? :wink: 

To sum up, if I am not mistaken:

Before this PR:
- The dropdown could be visually opened even though the colorpicker is not fully initialized yet (we did not wait for `_renderColorPalette`)
- But it could never be broken because we only relied on it after awaiting its initialization `this._colorPaletteRenderPromise` here
- => Any async operation is thus awaited in the editor mutex queue: we don't need to really await the rendering color palette in open as long as using it is awaited. `setValue` being part of the `updateUI` flow, which is awaiting on UI update, this is fine.

After this PR:
- The dropdown is now opened only when the colorpicker is ready. Arguably better: it may take a bit longer to open but the colorpicker is ready when it does.
- But when reaching this point in `setValue`, I don't think you have any guarantee that you can use it (`this.colorPaletteColorNames`). Basically this flow is probably broken:
  1. Select a value in a random option which does a long RPC
  2. Click on another colorpicker -> `_onClick` is called -> which calls `open`... whose results it not awaited anymore
  3. The RPC of (1) is finished, before the colorpicker is actually opened, an UI update is thus performed, which calls `setValue` on your colorpicker => bug

(I am doing this from memory so I might be wrong but the point is that I don't see anything that guarantees you can use the initialized colorpicker in `setValue` if you do not await the opening in some kind of mutexed queue ... which I don't think you can do)

Reply:
> why is that?

Because I need the color palette to be fully rendered to know its height and adjust the dropdown position.
Without waiting for it before `this._adjustDropdownPosition();` => it is empty.
I will try to understand how to do it differently following your comments.

Reply:
Yep, I understood that :wink:

Reply:
@qsm-odoo In the end, making `open()` async caused issues with certain options where we use `onClick()` on selectors.

For instance, when we activate the "on hover" animation:
-  We trigger a `click()` to choose a value in the selector.
- Then, we immediately call `close()` on the selector.
- Bug: the selector was no longer closed.

However, this no longer worked with asynchronous `open()`. This was due to the fact that the selector wasn't yet open when we attempted to close it. Therefore, I found a more reliable solution: waiting for the colorpalette before adjusting the position of the selector for the colorpicker.